### PR TITLE
feat(bazel): check all known index html file names for the index file

### PR DESCRIPTION
**Pull request by josephperrott**
Check for `index.csr.html`, `index.html` or `index.server.html` to serve the default index html file.

### CODE REVIEW

File: bazel/http-server/index.bzl [hunk 1]
```diff
@@ -80,6 +83,12 @@ http_server_rule = rule(
         "launcher": "%{name}.sh",
     },
     attrs = {
+        "serving_root": attr.string(
+            doc = """
+            The root path to serve files for the application from, defaults to the current
+            directory of the package.
```
Comment (devversion) [inline bazel/http-server/index.bzl:89]: OOC why is this needed? Isn't the serving root simply the directory where the rule is defined?

I would strongly prefer to preserve that "mental model" because all other access to "deps", or "assets" depends on that being the case as the "root dir".

Comment (devversion) [reply bazel/http-server/index.bzl:89]: This should also be already possible via `rootPaths` ?

Comment (josephperrott) [reply bazel/http-server/index.bzl:89]: In the case of an application created using architect, a `browser` and `server` directory are each created, and we further in framework put this into a `build` directory.

So we need to perform the serving for that project from `build/browser`

Comment (devversion) [reply bazel/http-server/index.bzl:89]: I think this can be achieved via `rootDirs` already?

Comment (josephperrott) [reply bazel/http-server/index.bzl:89]: Ahh I see it now, I was misunderstanding what it was looking for.

**Comment by josephperrott** [comment]
@devversion PTAL, I still would like to check for multiple possible index.html file names.

**Review by devversion — APPROVED** [review]

**Comment by josephperrott** [comment]
This PR was merged into the repository by commit 67489d145ae6769e0ca5408ecc91b6bbf8e56733.

The changes were merged into the following branches: main
